fix: fix error related to the exchanges addition and data about them

lukicenturi opened the pull request:
Closes #3862

## Checklist

- [ ] The PR modified the frontend, and updated the [user guide](https://github.com/rotki/rotki/blob/develop/docs/usage_guide.rst) to reflect the changes.
- [ ] Fix error related to the exchanges addition and purging and data related to them

yabirgb commented on frontend/app/src/store/history/actions.ts:
```diff
@@ -391,6 +422,23 @@ export const actions: ActionTree<HistoryState, RotkehlchenState> = {
+      limit: 1,
+      offset: 0,
+      ascending: false,
+      orderByAttribute: 'time'
```
Question: Isn't the sort field preserved when navigating between tabs? Is the sort by time here a problem?

kelsos replied:
These defaults are only for when the parameters are missing from the action.

The pagination event from the table will override these values.
Essentially these defaults are used only for refresh, where the result is ignored by the frontend, since the frontend is now using the `only_cache: true` (db) as a single source of truth.

kelsos commented on frontend/app/tests/e2e/pages/defi-page.ts:
```diff
@@ -19,9 +19,14 @@ export class DefiPage {
+      const moduleId = `#defi-module-${module}`;
+      cy.get('body').then($body => {
+        if ($body.find(moduleId).length > 0) {
+          cy.get(moduleId).scrollIntoView();
```
The same comment as the other PR applies here.

yabirgb commented:
So @lukicenturi I found and issue. For the setup I have already a kraken exchange instance and tried to add a FTX one. Added the exchange and navigated to trades. Then I see two queries to the same endpoint with the same parameters

![image](https://user-images.githubusercontent.com/5068010/147552535-c7be541a-e367-4583-822c-0e2347a1eab7.png)

And both query with `only_cache=true`

kelsos commented on frontend/app/src/store/history/mutations.ts:
```diff
@@ -27,13 +27,64 @@ export const mutations: MutationTree<HistoryState> = {
     state.trades = trades;
   },
 
+  [HistoryMutations.ADD_TRADE_LOCATION_TO_PURGE](
+    state: HistoryState,
+    location: TradeLocation
+  ) {
+    state.tradeLocationsToPurge = [...state.tradeLocationsToPurge, location];
+  },
+
+  [HistoryMutations.REMOVE_TRADE_LOCATION_TO_PURGE](
+    state: HistoryState,
+    location?: TradeLocation
+  ) {
+    if (!location) {
+      state.tradeLocationsToPurge = [];
+      return;
+    }
+
+    const locations = [...state.tradeLocationsToPurge];
+    const locationIndex = locations.indexOf(location);
+    if (locationIndex > -1) {
+      locations.splice(locationIndex, 1);
+      state.tradeLocationsToPurge = [...locations];
+    }
+  },
+
```
Wouldn't it be simpler to just trigger an `onlyCache: false` trade fetch when adding an exchange?

lukicenturi replied:
I thought about that too. But I thought of another scenario. 
If I just trigger `onlyCache: false` for a single exchange when adding an exchange, and then the user open the trade page, which will trigger `onlyCache: true` only, if the `onlyCache: false` before still not loaded, then when the user the open trade page, the data is still not updated. I decided to put it on a state, so if user open the trade page and there are `onlyCache: true` query that aren't done yet, just call it again.

lukicenturi replied:
But maybe it's too much. If you find it unnecessary, I will remove it

kelsos replied:
Let's remove it, we will re-evaluate the logic there either way

lukicenturi replied:
updated

kelsos commented on frontend/app/tests/e2e/pages/history-page/ledger-action-page.ts:
```diff
@@ -20,6 +20,7 @@ export class LedgerActionPage {
     return () => {
       // Wait for response.status to be 200
       cy.wait('@apiCall').its('response.statusCode').should('equal', 200);
+      cy.wait(500);
```
Is there not any alternative here other that hardcoding wait times?

lukicenturi replied:
I'm not sure, I can't find other easy way for fixing this. The problem is sometimes, the order of flow is like this:
call api -> cypress get some element -> the api changes is refresh whole table -> the previous element that cypress get is gone now.

lukicenturi replied:
> So @lukicenturi I found and issue. For the setup I have already a kraken exchange instance and tried to add a FTX one. Added the exchange and navigated to trades. Then I see two queries to the same endpoint with the same parameters
> 
> ![image](https://user-images.githubusercontent.com/5068010/147552535-c7be541a-e367-4583-822c-0e2347a1eab7.png)
> 
> And both query with `only_cache=true`. They both happen at the same time without me doing anything

Updated @yabirgb, can you try again?

lukicenturi replied:
Can we allow this for now and think the other way later? @yabirgb

kelsos commented on frontend/app/src/store/history/state.ts:
```diff
@@ -18,9 +18,12 @@ export const defaultState = (): HistoryState => ({
   transactions: defaultHistoricState<EthTransactionEntry>(),
-  ledgerActions: defaultHistoricState<LedgerActionEntry>()
+  ledgerActions: defaultHistoricState<LedgerActionEntry>(),
+  ledgerActionLocationsToPurge: []
```
In general, what I had in mind as an approach to this didn't require additions to the state, do you think we can take that approach here?

lukicenturi commented:
> So @lukicenturi I found and issue. For the setup I have already a kraken exchange instance and tried to add a FTX one. Added the exchange and navigated to trades. Then I see two queries to the same endpoint with the same parameters
> 
> ![image](https://user-images.githubusercontent.com/5068010/147552535-c7be541a-e367-4583-822c-0e2347a1eab7.png)
> 
> And both query with `only_cache=true`. They both happen at the same time without me doing anything

Updated @yabirgb, can you try again?

yabirgb reviewed (approved):
Related to functionality I approve this PR. If kelsos doesn't have any other comment we can GTM

kelsos reviewed (approved):
lgtm
